### DISCUSSION TITLE
Airflow architecture document of 2.0.0

### DISCUSSION THREAD

**Discussion by lixiaoyong12** [apache/airflow#13363] [Q&A] [answered]
Where can I view the architecture design document of 2.0.0?
Is the performance of 2.0.0 scheduler much due to previous versions?

**Comment by potiuk** [comment, marked as answer]
It's in the least expected place: http://airflow.apache.org/docs/apache-airflow/stable/start.html#basic-airflow-architecture

Yes, Performance of scheduler in 2.0 had improved a lot when it comes to speed of scheduling (but not the speed of executing tasks).  You can read more about the new scheduler implementation here: https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=103092651

**Reply by lixiaoyong12** [reply]
The scheduler can schedule the dag from the database instead of the *.py file.  because  of our dag is fixed .
